fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -320,7 +320,14 @@ public void copyOnlyForQuery(OlapTable olapTable) {
         olapTable.nameToColumn = Maps.newHashMap(this.nameToColumn);
         olapTable.state = this.state;
         olapTable.indexNameToId = Maps.newHashMap(this.indexNameToId);
-        olapTable.indexIdToMeta = Maps.newHashMap(this.indexIdToMeta);
+        olapTable.indexIdToMeta = Maps.newHashMap();
+        // fast schema evolution will modify indexIdToMeta in-place, hence we need shallow copy it here.
+        for (Map.Entry<Long, MaterializedIndexMeta> entry : indexIdToMeta.entrySet()) {
```

Comment:
could we change the fast schema evolution implementation?

Reply:
seems ok.

Reply:
One problem is someone may add a logic to modify some olapTable fields which may casue a new bug.